src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Rules/CannotChangeVisibility.cs

```diff
@@ -24,6 +25,40 @@ public CannotChangeVisibility(RuleSettings settings, IRuleRegistrationContext co
             context.RegisterOnTypeSymbolAction(RunOnTypeSymbol);
         }
 
+        private static Accessibility NormalizeInternals(Accessibility a) => a switch
+        {
+            Accessibility.ProtectedOrInternal => Accessibility.Protected,
+            Accessibility.ProtectedAndInternal or Accessibility.Internal => Accessibility.Private,
+            _ => a,
+        };
+
+        private int CompareAccessibility(Accessibility a, Accessibility b)
+        {
+            if (!_settings.IncludeInternalSymbols)
+            {
+                a = NormalizeInternals(a);
+                b = NormalizeInternals(b);
+            }
+
+            if (a == b)
+            {
+                return 0;
+            }
+
+            return (a, b) switch
+            {
+                (Accessibility.Public, _) => 1,
+                (_, Accessibility.Public) => -1,
+                (Accessibility.ProtectedOrInternal, _) => 1,
+                (_, Accessibility.ProtectedOrInternal) => -1,
+                (Accessibility.Protected or Accessibility.Internal, _) => 1,
+                (_, Accessibility.Protected or Accessibility.Internal) => -1,
+                (Accessibility.ProtectedAndInternal, _) => 1,
+                (_, Accessibility.ProtectedAndInternal) => -1,
+                _ => throw new NotImplementedException(),
```

Comment:
I'm a little bit worried about this line. This would cause a NotImplementedException being thrown when a new visibility is introduced by Roslyn. Is there a good default that we can provide for unknown accessibilities?

Also, if we would keep it, we would probably throw `ArgumentOutOfRangeException` instead.

Reply:
I'm not sure there is a default that applies to an `Accessibility` no matter where it is placed in the hierarchy. In particular, if it's an `Accessibility` like `Protected` or `Internal`, then it might special-casing, since all of this is a partial order.

`ArgumentOutOfRangeException` seems better suited for this scenario.

Reply:
I would follow the approach that we take in SymbolExtensions to not diverge in behavior: https://github.com/dotnet/sdk/blob/30dfe0a0726d0f9fd3ffb7c528fd1f0cc8ec6ec8/src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Extensions/SymbolExtensions.cs#L94

Reply:
Hmm, actually would it make sense to just do a `_ => a > b ? 1 : b > a ? -1 : 0` as the default for new unknown visibility members? Then you could also remove L43-45 as it would be handled by that default switch statement.